### PR TITLE
Don't copy annotations in MethodSpec.overriding

### PR DESCRIPTION
**Pull request by gk5885**
…for the reasons cited in issue #482.

### CODE REVIEW

File: src/main/java/com/squareup/javapoet/MethodSpec.java [hunk 1]
```diff
@@ -198,12 +201,7 @@ public static Builder overriding(ExecutableElement method) {
     String methodName = method.getSimpleName().toString();
     MethodSpec.Builder methodBuilder = MethodSpec.methodBuilder(methodName);
 
-    methodBuilder.addAnnotation(OVERRIDE);
```
Comment (sormuras) [inline src/main/java/com/squareup/javapoet/MethodSpec.java]: Why remove this? The result actually _does_ override a method.

Comment (ronshapiro) [inline src/main/java/com/squareup/javapoet/MethodSpec.java]: He added `methodBuilder.addAnnotation(Override.class)` which has the same effect.

Comment (sormuras) [inline src/main/java/com/squareup/javapoet/MethodSpec.java]: Ah! Missed that line. Never mind. :)

**Comment by swankjesse** [comment]
Merged manually.

**Pull request closed without merging**